lib/within_named_scope.rb

```diff
@@ -5,7 +5,7 @@ module WithinNamedScope
   end
 
   def in_scope?(scope)
-    !!self.class.send(scope.to_sym).exists?(self)
+    !!self.class.send(scope.to_sym).exists?(self.id)
```

Comment:
This could probably be `primary_key`, since it's not a guarantee the object is using `id`.